### PR TITLE
Make getWebSocketResponse polymorphic over monad

### LINKED ISSUE #4: Make a type alias for the requester constraint
`Reflex.Dom.WebSocket.Monad` should probably export `IsWebSocketRequest` and `IsWebSocketResponse`. Also handy would be a constraint to easily allow users to be polymorphic on it:

```
type HasWebSocket t msg m = Requester t (IsWebSocketRequest msg) (IsWebSocketResponse msg) m
```

### PR DESCRIPTION
**Pull request by 3noch**
Fixes #4

This allows you to use `getWebSocketResponse` in any monad except another `Requester` monad. That can be solved by using `newtype` and building all the instances. That can be done later.

Note that this change effectively deprecates `WithWebSocketT` and the examples need to change a bit. I can make those changes as well if you like.

### CODE REVIEW

**Comment by 3noch** [comment]
I apologize for all the diff noise. My editor is too ambitious 😠

File: reflex/Reflex/Dom/WebSocket/Monad.hs [hunk 1]
```diff
@@ -15,36 +16,41 @@
...
+import Reflex.Dom.Core hiding (Error, Value, WebSocket)
+import qualified Reflex.Dom.Core as Reflex.Dom
 
 import Control.Monad.State.Strict
 
 import Data.Aeson
 
 type WithWebSocketT ws t m = RequesterT t (IsWebSocketRequest ws) (IsWebSocketResponse ws) m
 
+type HasWebSocket t msg m = (Requester t m, Request m ~ IsWebSocketRequest msg, Response m ~ IsWebSocketResponse msg)
```
Comment (3noch) [inline reflex/Reflex/Dom/WebSocket/Monad.hs:52]: This is the new constraint we can use.

File: reflex/Reflex/Dom/WebSocket/Monad.hs [hunk 2]
```diff
@@ -66,8 +72,8 @@ data SomeRequesterDataKey ws where
     RequesterDataKey req -> SomeRequesterDataKey ws
 
 getWebSocketResponse
-  :: (WebSocketMessage ws req, Monad m, Reflex t)
-  => Event t req -> WithWebSocketT ws t m (Event t (ResponseT ws req))
+  :: (WebSocketMessage ws req, HasWebSocket t ws m)
+  => Event t req -> m (Event t (ResponseT ws req))
```
Comment (3noch) [inline reflex/Reflex/Dom/WebSocket/Monad.hs:76]: And now we're using `m` with a constraint.

**Comment by dfordivam** [comment]
Good for now...
Though I do not like the idea of exporting IsWebSocket* stuff

**Comment by 3noch** [comment]
@dfordivam I'm curious why you don't like exporting those?

**Comment by dfordivam** [comment]
Its just an internal stuff ... so I wanted the interface to not export those..
May be no harm even if it is exported..

**Pull request merged**